Tests/base/NSData/search.m

```diff
@@ -0,0 +1,81 @@
+#import <Foundation/Foundation.h>
+
+#import "Testing.h"
+
+static BOOL rangesEqual(NSRange r1, NSRange r2)
```

Comment:
You could use NSEqualRanges instead.

Reply:
I can certainly do that, it's just what the NSString/test00.m did, so I wasn't sure if there was a reason for a special function.

Reply:
The two things are subtly different.
NSEqualRanges checks to see if two ranges both have the same length and location,  but the function allows two ranges of zero length to be counted as equal.  I suspect that this comes from days when OSX would consistently return a zero length range when a string was not found, but would leave the location indeterminate.
I think using NSEqualRanges is preferable, as long as the testcase pass on OSX.